presto-main/src/main/java/com/facebook/presto/execution/SqlStageExecution.java

```diff
@@ -549,7 +598,12 @@ private synchronized RemoteTask scheduleTask(InternalNode node, TaskId taskId, M
 
     public Set<InternalNode> getScheduledNodes()
     {
-        return ImmutableSet.copyOf(tasks.keySet());
+        if (isEnableGracefulShutdown || isRetryOfFailedSplitsEnabled) {
```

Comment:
I am wondering if this compound condition is a smell ? We need to specify it so many places and we can mess it up. How can we improve it ? Like a helper method perhaps or such ?